Libraries/Components/View/View.js

```diff
@@ -27,12 +27,94 @@ const View: React.AbstractComponent<
   ViewProps,
   React.ElementRef<typeof ViewNativeComponent>,
 > = React.forwardRef(
-  ({tabIndex, focusable, ...otherProps}: ViewProps, forwardedRef) => {
+  (
+    {tabIndex, focusable, role, accessibilityRole, ...otherProps}: ViewProps,
+    forwardedRef,
+  ) => {
+    let restProps = {...otherProps};
+
+    // Map role values to AccessibilityRole values
+    const roleToAccessibilityRoleMapping = {
+      alert: 'alert',
+      button: 'button',
+      checkbox: 'checkbox',
+      combobox: 'combobox',
+      grid: 'grid',
+      heading: 'header',
+      img: 'image',
+      link: 'link',
+      list: 'list',
+      menu: 'menu',
+      menubar: 'menubar',
+      menuitem: 'menuitem',
+      none: 'none',
+      presentation: 'none',
+      progressbar: 'progressbar',
+      radio: 'radio',
+      radiogroup: 'radiogroup',
+      scrollbar: 'scrollbar',
+      searchbox: 'search',
+      slider: 'adjustable',
+      spinbutton: 'spinbutton',
+      summary: 'summary',
+      switch: 'switch',
+      tab: 'tab',
+      tablist: 'tablist',
+      timer: 'timer',
+      toolbar: 'toolbar',
+      tooltip: undefined,
+      feed: undefined,
+      math: undefined,
+      note: undefined,
+      application: undefined,
+      article: undefined,
+      cell: undefined,
+      columnheader: undefined,
+      definition: undefined,
+      directory: undefined,
+      document: undefined,
+      figure: undefined,
+      group: undefined,
+      listitem: undefined,
+      meter: undefined,
+      row: undefined,
+      rowgroup: undefined,
+      rowheader: undefined,
+      separator: undefined,
+      table: undefined,
+      term: undefined,
+      tabpanel: undefined,
+      treeitem: undefined,
+      tree: undefined,
+      treegrid: undefined,
+      banner: undefined,
+      complementary: undefined,
+      contentinfo: undefined,
+      form: undefined,
+      main: undefined,
+      navigation: undefined,
+      region: undefined,
+      log: undefined,
+      marquee: undefined,
+      status: undefined,
+      alertdialog: undefined,
+      dialog: undefined,
+    };
+
+    const _accessibilityRole = role
+      ? roleToAccessibilityRoleMapping[role]
+      : accessibilityRole;
+
+    // set restProps is _accessibilityRole exists
+    if (_accessibilityRole) {
+      restProps = {...restProps, accessibilityRole: _accessibilityRole};
+    }
```

Comment:
We shouldn't need to merge this into restProps as `accessibilityRole` can be defined below as `focusable` was in earlier diffs